pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -119,79 +118,58 @@ type ResultFromCRCreate struct {
-	c := make(chan ResultFromCRCreate)
-	var wg sync.WaitGroup
+	results := []ResultFromCRCreate{}
 
 	resources := r.getAllResources(cr, request)
 	for _, desiredRuntimeObj := range resources {
-		wg.Add(1)
-		go func(desiredRuntimeObj runtime.Object) {
-			defer wg.Done()
-			desiredMetaObj := desiredRuntimeObj.(metav1.Object)
-
-			// use reflection to create default instance of desiredRuntimeObj type
-			typ := reflect.ValueOf(desiredRuntimeObj).Elem().Type()
-			currentRuntimeObj := reflect.New(typ).Interface().(runtime.Object)
-
-			key := client.ObjectKey{
-				Namespace: desiredMetaObj.GetNamespace(),
-				Name:      desiredMetaObj.GetName(),
-			}
-			err := r.client.Get(context.TODO(), key, currentRuntimeObj)
+		desiredMetaObj := desiredRuntimeObj.(metav1.Object)
 
-			if err != nil {
-				if !errors.IsNotFound(err) {
-					c <- ResultFromCRCreate{result: reconcile.Result{}, err: err}
-				}
+		// use reflection to create default instance of desiredRuntimeObj type
+		typ := reflect.ValueOf(desiredRuntimeObj).Elem().Type()
+		currentRuntimeObj := reflect.New(typ).Interface().(runtime.Object)
 
-				if err = controllerutil.SetControllerReference(cr, desiredMetaObj, r.scheme); err != nil {
-					c <- ResultFromCRCreate{result: reconcile.Result{}, err: err}
-				}
+		key := client.ObjectKey{
+			Namespace: desiredMetaObj.GetNamespace(),
+			Name:      desiredMetaObj.GetName(),
+		}
+		err := r.client.Get(context.TODO(), key, currentRuntimeObj)
 
-				// TODO: common-templates and cdi fails the Get check above and appears to still be missing.
-				// But in reality, when you try to Create it, the client reports back that
-				// the resource already exists. Need to investigate why.
-				// Before the refactor, the code didn't check if a resource already exists. It just
-				// tried to create it, and will skip if the Create indicated that it already exists.
-				if err = r.client.Create(context.TODO(), desiredRuntimeObj); err != nil {
-					if err != nil && errors.IsAlreadyExists(err) {
-						logger.Info("Skip reconcile: tried create but resource already exists", "key", key)
-						c <- ResultFromCRCreate{result: reconcile.Result{}, err: nil}
-					} else if err != nil {
-						c <- ResultFromCRCreate{result: reconcile.Result{}, err: err}
-					}
-				} else {
-					logger.Info("Resource created",
-						"namespace", desiredMetaObj.GetNamespace(),
-						"name", desiredMetaObj.GetName(),
-						"type", fmt.Sprintf("%T", desiredMetaObj))
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				results = append(results, ResultFromCRCreate{result: reconcile.Result{}, err: err})
+			}
+
+			if err = controllerutil.SetControllerReference(cr, desiredMetaObj, r.scheme); err != nil {
+				results = append(results, ResultFromCRCreate{result: reconcile.Result{}, err: err})
+			}
+
+			// TODO: common-templates and cdi fails the Get check above and appears to still be missing.
```

Comment:
I noticed in the past that the code that queries for CDI fails, unless an empty string ("") is used for the namespace. It may be related.

Reply:
I like the flow of check and if not exist, create. It would be good to figure out why the GET requests aren't working correctly. @rthallisey, do you know where they are globally scoped? 

thanks @lveyde, I checked and setting namespace to "" for CDI fixes it. Is empty string ok as the namespace for CDI?

Need to figure out what is going on with common-templates.

Reply:
Ya an empty string is how you're suppose to reference a globally scoped CRD.  We'll need keep a list of those and have different check for them. 

>@rthallisey, do you know where they are globally scoped?
>
Not sure I understand the question.

The empty string didn't also GET the common-templates?

Reply:
@rthallisey let me rephrase, how do you know if something is globally scoped?

The empty string doesn't work for common-templates.

Reply:
If the something you are referring to is a CRD. Then you can look at the `spec.scope` of the CRD.

For example, [cluster network addons](https://github.com/kubevirt/hyperconverged-cluster-operator/blob/master/deploy/converged/crds/cna.crd.yaml#L13) or [kubevirt](https://github.com/kubevirt/hyperconverged-cluster-operator/blob/master/deploy/converged/crds/kubevirt.crd.yaml#L24)